ctapipe_io_magic/laser_params_corr_test.py

```diff
@@ -0,0 +1,411 @@
+import logging
+import datetime
+from pathlib import Path
+import os
+import numpy as np
+import astropy.units as u
+from ctapipe_io_magic import MAGICEventSource
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+class ReportLaser:
+    def magic_reports_reading(self, input_file, process_run=False):
+        logger.info(f"\nInput file: {input_file}")
+        event_source = MAGICEventSource(input_file, process_run=process_run)
+        is_simulation = event_source.is_simulation
+        logger.info(f"\nIs simulation: {is_simulation}")
+
+        obs_id = event_source.obs_ids[0]
+        tel_id = event_source.telescope
+
+        logger.info(f"\nObservation ID: {obs_id}")
+        logger.info(f"Telescope ID: {tel_id}")
+
+        laser = event_source.laser
+
+        for key, report_list in laser.items():
+            for report in report_list:
+                logger.info(f"Accessing parameters for each ReportLaserContainer object")
+
+        return report
+
+    def get_c0(self, time, zenith, range_max):
+        filename = '/home/zywuckan/ctapipe_io_magic/ctapipe_io_magic/params.txt'
+
+        gkC_0Period1 = None
+        gkFullOverlap5 = None
+        gkFullOverlap6 = None
+        gkFullOverlap7 = None
+        gkIntegrationWindow = None
+        gkHWSwitchV1to4 = None
+
+        with open(filename, 'r') as file:
+            for line in file:
+                if line.strip():
+                    variable, value = line.split("=")
+                    variable = variable.strip()
+                    value = value.split("#")[0].strip()
+                    if variable == "stime21":
```

Comment:
all this code just to initialize a bunch of variables to -1 ??
and you should use arrays of parameters rather then stimeXX